From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harmer <mark.harmer@discordapp.com>
Date: Mon, 27 Feb 2023 12:56:15 -0800
Subject: Enabling AudioVideoTracks feature to use HTMLMediaElements
 audioTracks.

enables audiotrack feature

diff --git a/third_party/blink/renderer/platform/runtime_enabled_features.json5 b/third_party/blink/renderer/platform/runtime_enabled_features.json5
index ce3c7fa860bae9b5946105b05b0e7618bd658d17..9b0ed1a33c28886bdd76a4ce0f49a564c3c9a4bb 100644
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -361,7 +361,7 @@
     },
     {
       name: "AudioVideoTracks",
-      status: "experimental",
+      status: "stable",
     },
     {
       name: "AutoDarkMode",
